### ISSUE TITLE
Resume a suspended cronjob will trigger workflow twice

### ISSUE THREAD

**Issue by monsterxx03** [argoproj/argo#4445]
## Summary 

Create a cronjob run at every minute. Let it scheduled once(so LastScheduledTime will be filled). Then:

`argo cron suspend test-cron-wf`

Wait for 1 min, after  its next scheduled time, do:

`argo cron resume test-cron-wf`

I will see two workflows are created.

`argo list:`

    test-cron-wf-rc45q                                 Running     4s    4s         0
    test-cron-wf-gw8t2                                Running     4s    4s         0

I only have one argo workflow controller in cluster, it's 100% reproducible in my case. 

## Diagnostics

What Kubernetes provider are you using? 

aws eks 1.16

What version of Argo Workflows are you running? 

2.11.6

```yaml
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
  name: test-cron-wf
spec:
  schedule: "* * * * *"
  concurrencyPolicy: "Forbid"
  workflowSpec:
    entrypoint: whalesay
    templates:
    - name: whalesay
      container:
        image: alpine:3.6
        command: [sh, -c]
        args: ["date; sleep 20"]
```

Logs
```
time="2020-11-03T06:08:09Z" level=info msg="Processing glow/test-cron-wf" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="test-cron-wf missed an execution at Tue Nov  3 06:08:00 2020 and is within StartingDeadline" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Running test-cron-wf" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:09Z" level=info msg="Updated phase  -> Running" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:09Z" level=info msg="CronWorkflow glow/test-cron-wf added" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Processing glow/test-cron-wf" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="CronWorkflow glow/test-cron-wf added" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Pod node test-cron-wf-rc45q initialized Pending" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:09Z" level=info msg="Created pod: test-cron-wf-rc45q (test-cron-wf-rc45q)" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:09Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204694 workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:09Z" level=info msg="Processing glow/test-cron-wf" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="test-cron-wf missed an execution at Tue Nov  3 06:08:00 2020 and is within StartingDeadline" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Running test-cron-wf" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:09Z" level=info msg="Updated phase  -> Running" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:09Z" level=info msg="CronWorkflow glow/test-cron-wf added" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Processing glow/test-cron-wf" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="CronWorkflow glow/test-cron-wf added" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Pod node test-cron-wf-gw8t2 initialized Pending" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:09Z" level=info msg="Created pod: test-cron-wf-gw8t2 (test-cron-wf-gw8t2)" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:09Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204704 workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:09Z" level=info msg="Processing glow/test-cron-wf" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="CronWorkflow glow/test-cron-wf added" cronWorkflow=glow/test-cron-wf
time="2020-11-03T06:08:09Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf

time="2020-11-03T06:08:10Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:10Z" level=info msg="Updating node test-cron-wf-rc45q message: ContainerCreating"
time="2020-11-03T06:08:10Z" level=info msg="Skipped pod test-cron-wf-rc45q (test-cron-wf-rc45q) creation: already exists" namespace=glow podPhase=Pending workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:10Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204719 workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:10Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf

time="2020-11-03T06:08:10Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:10Z" level=info msg="Updating node test-cron-wf-gw8t2 message: ContainerCreating"
time="2020-11-03T06:08:10Z" level=info msg="Skipped pod test-cron-wf-gw8t2 (test-cron-wf-gw8t2) creation: already exists" namespace=glow podPhase=Pending workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:10Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204722 workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:10Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf

time="2020-11-03T06:08:10Z" level=info msg="Alloc=37585 TotalAlloc=355890379 Sys=144258 NumGC=19898 Goroutines=182"
time="2020-11-03T06:08:11Z" level=info msg="Performing periodic workflow GC"
time="2020-11-03T06:08:11Z" level=info msg="Zero old offloads, nothing to do"
time="2020-11-03T06:08:11Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:11Z" level=info msg="Updating node test-cron-wf-rc45q status Pending -> Running"
time="2020-11-03T06:08:11Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204734 workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:11Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:11Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:11Z" level=info msg="Skipped pod test-cron-wf-gw8t2 (test-cron-wf-gw8t2) creation: already exists" namespace=glow podPhase=Pending workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:12Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-rc45q
time="2020-11-03T06:08:13Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:13Z" level=info msg="Updating node test-cron-wf-gw8t2 status Pending -> Running"
time="2020-11-03T06:08:13Z" level=info msg="Enforcing history limit for 'test-cron-wf'" namespace=glow workflow=test-cron-wf
time="2020-11-03T06:08:13Z" level=info msg="Workflow update successful" namespace=glow phase=Running resourceVersion=271204750 workflow=test-cron-wf-gw8t2
time="2020-11-03T06:08:14Z" level=info msg="Processing workflow" namespace=glow workflow=test-cron-wf-gw8t2
```
From the log, I can see  `test-cron-wf missed an execution at Tue Nov  3 06:08:00 2020 and is within StartingDeadline` triggered twice, looks like a race condition.

BTW, I have another issue similar with #2773,I run argo for about 2 weeks, some of my cronjobs(maybe daily/weekly) will create two workflows at the scheduled time(I set `concurrencyPolicy: Forbid` for all),  very like this suspend/resume issue. But not every often, in two weeks I come to 3 times(happened to random daily cronjob). Since I haven't collect useful log, just list it here

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by monsterxx03** [comment]
Maybe  a duplicated issue of https://github.com/argoproj/argo/issues/4392

**Comment by simster7** [comment]
Yes, this should be fixed with #4440. Closing as duplicate. Thanks for the detailed bug report anyway

**Comment by simster7** [comment]
Just confirmed this is fixed with #4440

**Comment by monsterxx03** [comment]
I upgraded to 2.12.0-rc1 yesterday, the suspend/resume  trigger wf twice issue go away. 

But I still come to cronwf trigger twice on scheduled time today. After read #4392 again, I think the reason is I use kubectl apply -f xxx.yaml to update cronwf. It's using PATCH internally. 

So the right way to update cronwf from cli is delete & create again?  I checked argo server code, seems it's using UPDATE.

I think argo should clarify in doc not using `kubectl apply`, and support `argo cron update`
